controllers/workspace.js

```diff
@@ -4,24 +4,84 @@
 
 'use strict';
 
+const _url = require('url');
+
 const Handler = require('../lib/handler');
 
 const workspace_factory = require('../assetmanager/workspace_factory');
 const _repo_manager = require('../assetmanager/repo_manager');
 const utilities = require('../assetmanager/utilities');
 const _workspace_metadata_presenter = require('../assetmanager/workspace_presenter').Workspace_metadata_presenter;
+const Workspace_presenter = require("../assetmanager/workspace_presenter").Workspace_presenter;
+
+const list_workspaces_regexp = /^\/contentbrowser\/workspaces(\/)?$/;
+const get_workspaces_regexp = /^\/contentbrowser\/workspaces\/([^/]*)$/;
 
 const workspaces_regexp = /^\/assetmanager\/workspaces\/([^/]*)$/;
 const workspace_reset_regexp = /^\/assetmanager\/workspaces\/([^/]*)\/reset$/;
-const favorites_regexp = /^\/assetmanager\/workspaces\/([^/]*)\/favorites$/;
 
 const errors = require('../lib/errors')('asset manager');
 
+// Utility for slicing a list of files
+const slice = (files, start, number) => files.slice(start, start + number);
+
+// Utility to create the next reference url for paging
+const get_next_url = (url, next) => {
+    let parsed_url = _url.parse(url, true);
+    parsed_url.query.start = next;
+    parsed_url.search = undefined;  // If search is set it prevails over the query, we must unset it
+    return parsed_url.format();
+};
+
 module.exports = function(server, context) {
     const favorite = context.favorite;
     const _project = require('../assetmanager/project_manager')(context);
     const _workspace = require('../assetmanager/workspace')(context);
 
+    server.get(list_workspaces_regexp, function(req, res, next) {
+        const handler = new Handler(req, res, next);
+        const options = {
+            filterName: req.query.name,
+            maxResults: parseInt(req.query.maxResults, 10) || 100,
+            start: parseInt(req.query.start, 10) || 0
+        };
+        _workspace
+            .list(options)
+            .then(workspaces => {
+                const length = workspaces.length;
+                workspaces = slice(workspaces, options.start, options.maxResults);
+                const output = {
+                    kind: 'workspace-list',
+                    items: workspaces.map(workspace => Workspace_presenter.present(workspace))
+                };
+                if (options.start + workspaces.length < length) {
+                    const indexOfMoreResults = options.start + options.maxResults;
+                    output.nextLink = get_next_url(req.url, indexOfMoreResults);
+                }
+                output.totalItems = length;
+                handler.sendJSON(output, 200);
+            })
+            .catch(err => handler.handleError(err));
+    });
+
+    server.get(get_workspaces_regexp, function(req, res, next) {
```

Comment:
If it is really to get one workspace then the regexp should be named in singular
```suggestion
    server.get(get_workspace_regexp, function(req, res, next) {
```